### PR TITLE
[core] Constant use element iterator

### PR DESCRIPTION
**Pull request by praasz**
### Details:
- Add LP specializations for cast, fill, write data.
- Remove not used private members.
- Move public non-template members to cpp.
- Optimize bin-size (Ubuntu 20, Release, gcc11):
  - No LTO around 421 KiB
  - LTO around  10.6 KiB
- Add common elements byte size util function.
- Prepare to use new type like u2, u3, u6

### Tickets:
 - Part of [CVS-119224](https://jira.devtools.intel.com/browse/CVS-119224)

### CODE REVIEW

File: src/core/include/openvino/op/constant.hpp [hunk 1]
```diff
@@ -503,102 +436,50 @@ class OPENVINO_API Constant : public Op {
...
 
     template <element::Type_t Type,
-              typename T,
-              typename StorageDataType = fundamental_type_for<Type>,
-              typename std::enable_if<Type != element::Type_t::u1 && Type != element::Type_t::u4 &&
-                                          Type != element::Type_t::i4 && Type != element::Type_t::nf4 &&
-                                          Type != element::Type_t::string,
-                                      bool>::type = true>
+              class T,
+              typename std::enable_if<(!is_lp_type(Type) && Type != element::string) &&
```
Comment (t-jankowski) [inline src/core/include/openvino/op/constant.hpp:479]:
```suggestion
              typename std::enable_if<!is_lp_type(Type) && Type != element::string &&
```

Comment (praasz) [reply src/core/include/openvino/op/constant.hpp]: Changed.

File: src/core/src/op/constant.cpp [hunk 2]
```diff
@@ -117,28 +176,36 @@ Constant::Constant(bool memset_allocation, const element::Type& type, const Shap
...
     } else {
-        std::memcpy(get_data_ptr_nc(), data, mem_size());
+        std::memcpy(get_data_ptr_nc(), data, element::get_byte_size(m_element_type, num_elements));
     }
 }
 
+Constant::Constant(const element::Type& type, const Shape& shape, const std::shared_ptr<ov::AlignedBuffer>& data)
+    : m_element_type(type),
+      m_shape(shape),
+      m_data{data} {
```
Comment (t-jankowski) [inline src/core/src/op/constant.cpp:205]: Mixed style - `()` vs `{}`

**Review by t-jankowski — APPROVED** [review]

File: src/core/src/op/constant.cpp (hunk 2 above)
Comment (praasz) [reply src/core/src/op/constant.cpp:205]: If there will any changes to this PR will change it or change with new PR where new types will be added.

**Comment by mlukasze** [comment]
@ilya-lavrenov && @itikhono could you take a look, please?

**Review by vurusovs — APPROVED** [review]

**Pull request merged**